### ISSUE TITLE
no-return-await rule is depraceted since node 10

### ISSUE THREAD

**Issue by czlowiek488** [eslint/eslint#14613]
**What rule do you want to change?**
no-return-await

**Does this change cause the rule to produce more or fewer warnings?**
Remove warning and add different, maybe depend on used node version.
Or even maybe remove this rule and add different?

**How will the change be implemented? (New option, new default behavior, etc.)?**
New default behaviour, should depend on used node version.

**Please provide some example code that this change will affect:**

This code will result with different output depending on which node version you are using. In the picture I show result with node 12 but this change was introduced in node 10.
```js
const sleep = delay => new Promise(resolve => setTimeout(resolve, delay));
const test1 = async (a = 0) => {
    await sleep(10);
    if(a < 10) return test1(a+1);
    else console.trace('noawait', a);
};
test1();
const test2 = async (a = 0) => {
    await sleep(10);
    if(a < 10) return await test2(a+1);
    else console.trace('await', a);
};
test2();
```
![image](https://user-images.githubusercontent.com/34620109/119222214-7ce2f500-baf3-11eb-8cd5-03445f3a150a.png)

**What does the rule currently do for this code?**
It highlight error if there is `return await promise`.
**What will the rule do after it's changed?**
It should highlight error if there is `return promise` instead.
**Are you willing to submit a pull request to implement this change?**
No

**Comment by czlowiek488** [comment]
This rule was introduced at [2016-10-10](https://github.com/eslint/eslint/commit/66fe9ffa72487343fd2e21673b79281424cff6ef) and node 10 was introduced at [2018-04-24](https://nodejs.org/download/release/v10.0.0/).

**Comment by nzakas** [comment]
Thanks for the issue. I think you’re trying to make the point that there may be a situation when you don’t want to use this rule. However, we do not change rule behavior based on the environment the rule is being used in. JavaScript code can be executed in many different runtimes, each with their own behaviors, and so we don’t make rules specific to any one runtime. (You can do that if you would like by creating a custom rule.)
